### ISSUE TITLE
Option to add extra policy metadata

### ISSUE THREAD

**Issue by JohnHillegass** [cloud-custodian/cloud-custodian#5968]
**Is your feature request related to a problem? Please describe.**
It is sometimes hard to pin down who created a policy document using version control when it has been edited many times. It is also hard to track the creation date, document versioning, linkage to approval documents, etc. 

**Describe the solution you'd like**
I'd like to add a key to the schema called `metadata` where we could store some of that information

**Describe alternatives you've considered**
I tried using tags but since they are an array of strings, it does not easily allow for more complex mapping or inclusion of numbers without parsing. I think having a policy tag dedicated to this would be a nice feature to have. 

```
policies:
  - name: app-test
     resource: app-elb
     metadata:
       version: 1.0
       policyAuthor: Some One
       createdDate: 07/09/20 11:31
       lastModified: 07/18/20 09:21
       designedBy: Other One
       designLink: www.not_a_real_link.com
       designReviewedBy: The One
    tags: 
      - 'risk:medium'
      - 'designID:0001-D'
```

**Additional context**

**Comment by kapilt** [comment]
sounds good to me. although the delta to k:v labels is only meaningful if we have extended basic types here (bool, integer, etc).

**Comment by JohnHillegass** [comment]
Closed by #5969
